### DISCUSSION TITLE
Selecting source file in importer

### DISCUSSION THREAD

**Discussion by Mastomaki** [Spine-project/Spine-Toolbox#2543] [Support]
In importer specification, is it possible to select the source file, if the importer has several available resources? The selection drop down box is greyed. Now I have to go back to the main toolbox window to select the source file (defaults to the first one in the list?).

![Screenshot_20240202_102209](https://github.com/spine-tools/Spine-Toolbox/assets/42839563/a3610e62-fb0c-48c5-b537-d5d4c5d1db18)

**Comment by soininen** [comment]
Can your multiple source files all be imported using the same import mappings? If not, then you should have a separate importer for each one of them.

The Source files selection was originally meant to be used in case you are building import mappings without an Importer item, e.g. when you want to make a specification for a plugin. Maybe it could be considered a bug that it is disable (grayed out) when you are editing the specification of an existing Importer. Most users just have a single file there (or datapackage in case of multiple csv files) so the issue has never come up I guess.

**Comment by Mastomaki** [comment]
>Can your multiple source files all be imported using the same import mappings? If not, then you should have a separate importer for each one of them.

No. I guess I should make separate importers.
